src/openrct2-ui/windows/NewRide.cpp

```diff
@@ -286,6 +292,10 @@ class NewRideWindow final : public Window
         PopulateRideList();
         InitScrollWidgets();
 
+        _filter.resize(MAX_PATH);
+        widgets[WIDX_FILTER_TEXT_BOX].string = _filter.data();
```

Comment:
Instead of doing this set the  `widgets[WIDX_FILTER_TEXT_BOX].string` in the `OnPrepareDraw` function. That is guaranteed to be called prior to drawing so the pointer will be valid at time of drawing. Previously we used to do dynamic strings via a similar method with the common format args but its also pretty messy.